src/TRoomDB.cpp

```diff
@@ -715,23 +715,21 @@ void TRoomDB::auditRooms(QHash<int, int>& roomRemapping, QHash<int, int>& areaRe
             // Sort the ids so that the reporting is ordered, which could be
             // helpful if there is a large number of faults
             std::sort(missingAreasNeeded.begin(), missingAreasNeeded.end());
-            for (int newAreaId : missingAreasNeeded) {
-
-                // This will create a new "Default" area name if there is not one
-                // already for this id - and we do not anticipate that it could ever
-                // fail and return false...
-                addArea(newAreaId);
-                infoMsg.append(QStringLiteral("\n%1 ==> \"%2\"")
-                               .arg(newAreaId)
-                               .arg(areaNamesMap.value(newAreaId)));
-            }
+        }
 
-            // Didn't really needed to be done, but as we have finished with it
-            // now, clearing it may make tracking the overall processes going on
-            // in the debugger a little clearer...
-            missingAreasNeeded.clear();
+        for (int newAreaId : missingAreasNeeded) {
```

Comment:
I only spotted this logic error when I had a map with a __single__ instance of an area name entry but no rooms with that area and no `TArea` instance - the issue was spotted and reported but because this bit of code was inside the `if (moreThanOneMissingTArea) sortThemByIdNumbersFirstBeforeAddingEachOne` code it was not being done - repeatedly... 🙁